### ISSUE TITLE
jasmine 3.0+ doesn't work

### ISSUE THREAD

**Issue by danilaml** [jasmine/jasmine-py#52]
For some reason, when I try to run `jasmine ci` on a simple test spec I get

```
[24/Jul/2018:17:11:11] ENGINE Bus STARTING
[24/Jul/2018:17:11:11] ENGINE Started monitor thread 'Autoreloader'.
[24/Jul/2018:17:11:11] ENGINE Serving on http://127.0.0.1:63941
[24/Jul/2018:17:11:11] ENGINE Bus STARTED
[24/Jul/2018:17:11:24] ENGINE Bus STOPPING
[24/Jul/2018:17:11:24] ENGINE HTTP Server cherrypy._cpwsgi_server.CPWSGIServer(('127.0.0.1', 63941)) shut down
[24/Jul/2018:17:11:24] ENGINE Stopped thread 'Autoreloader'.
[24/Jul/2018:17:11:24] ENGINE Bus STOPPED
usage: jasmine [-h] {server,ci,init} ...

Jasmine command line

positional arguments:
  {server,ci,init}  commands
    server          Jasmine server
    ci              Jasmine CI
    init            initialize Jasmine

optional arguments:
  -h, --help        show this help message and exit
c:\python27\lib\site-packages\cherrypy\process\wspbus.py:258: RuntimeWarning: The main thread is exiting, but the Bus is in the states.STOPPED state; shutting it down automatically now. You must either call bus.block() after start(), or call bus.exit() before the main thread exits.
  'main thread exits.' % self.state, RuntimeWarning)
[24/Jul/2018:17:11:24] ENGINE Bus STOPPING
[24/Jul/2018:17:11:24] ENGINE HTTP Server cherrypy._cpwsgi_server.CPWSGIServer(('127.0.0.1', 63941)) already shut down
[24/Jul/2018:17:11:24] ENGINE No thread running for None.
[24/Jul/2018:17:11:24] ENGINE Bus STOPPED
[24/Jul/2018:17:11:24] ENGINE Bus EXITING
[24/Jul/2018:17:11:24] ENGINE Bus EXITED
```
Notice the jasmine help in the middle. The same happens regardless of the browser (ie, firefox, phatomjs) or OS (windows, Linux). If I install jasmine `2.9` or `2.99`, then it runs test specs as it should.
Not sure what's wrong and where to look. Don't see anything suspicious in the logs.

**Comment by slackersoft** [comment]
It looks like the way Jasmine is trying to fall back to printing the cli help is covering up whatever the real cause is. Jasmine should probably look at doing a check for finding a known command instead of doing an `except AttributeError`.

**Comment by danilaml** [comment]
@slackersoft any way for me to debug this/improve the output? My setup is rather basic so I'm not sure what might've caused this.

**Comment by slackersoft** [comment]
Try 3.1.2 (just released) which shouldn't be covering over any exceptions at the CLI level anymore. This should point you toward the exception that is causing Jasmine to crash. If it looks like it's still an issue with Jasmine, please open up a new issue with the updated output, or post it back here.

Thanks!

**Comment by danilaml** [comment]
@slackersoft my apologies. It looks like I've missed that 3.0+ has dropped the support for py2.7 (I've seen it in the tox file and assumed it's still supported). Any reason to do this in particular? 

2.7 is still supported and widely used on some distributions (for at least the next 2 years). It looks like the only real change required to support it is replacing `import urlib` with `import six.moves.urllib as urllib` in a few places (at least I've managed to run the `jasmine ci` just by changing one line here https://github.com/jasmine/jasmine-py/blob/master/jasmine/url_builder.py#L1).

**Comment by slackersoft** [comment]
We dropped support for python 2.x for a combination of reasons.

Support for python 2.6 & 2.7 (especially on windows) was taking up the bulk of the time we had to spend maintaining all of Jasmine. This was partially from some of the dependencies moving away from support and partially due to the difficulty we were having keeping and getting development environments setup to work with the different versions of Python.

Additionally, we only want to drop support for platforms in a major release version of Jasmine. Given the current release cadence and time to spend on Jasmine, we didn't want to leave ourselves in the situation of supporting older versions of python until Jasmine 4.0.

Hope this helps.
